apps/src/lab2/views/dialogs/ShareDialog.tsx

```diff
@@ -165,6 +171,14 @@ const ShareDialog: React.FunctionComponent<{
     );
   }, [channelId, dispatch, projectType]);
 
+  const feedbackLink = useAppSelector(state => {
+    const {userType, signInState} = state.currentUser;
+    if (signInState !== SignInState.SignedIn) return undefined;
+    return userType === 'teacher'
+      ? TEACHER_FEEDBACK_LINK
+      : STUDENT_FEEDBACK_LINK;
+  });
```

Comment:
Ideally we would be able to unify this code (and the links above) with the same code in https://github.com/code-dot-org/code-dot-org/pull/62707.  That said, it might be too risky for this close to HOC.

Would value @sanchitmalhotra126 and @molly-moen's thoughts on a good architecture for each lab to influence the congrats dialog that is being shown, while keeping the lab and congrats dialog code independent.  We could also use such a facility for the other Music Lab customization to the congrats panel: the AFE section.

Reply:
I've detailed a follow-up task to unify the code/links above, but I agree, probably too risky for this close to HOC

Reply:
Yep 100%, didn't realize this code was doing to be in two different dialogs. This also technically means that any lab2 progression with a finish dialog and a finish URL will see these feedback buttons, not just Music Lab (though Music Lab HOC is currently the only lab2 progression with a finish dialog). Seems okay for now, but would definitely be ideal to revisit this post HOC 🙂 thanks for filing the task!

My initial thought is that any metadata like this (feedback links, dialog style) can be provided as part of level properties so we can configure this on a per-level basis, but we can definitely revisit post HOC.